src/test/java/core/basesyntax/strategy/operation/OperationHandlersTest.java

```diff
@@ -0,0 +1,94 @@
+package core.basesyntax.strategy.operation;
+
+import static core.basesyntax.strategy.OperationHandlersUtil.BALANCE_OPERATION_HANDLER;
+import static core.basesyntax.strategy.OperationHandlersUtil.PURCHASE_OPERATION_HANDLER;
+import static core.basesyntax.strategy.OperationHandlersUtil.RETURN_OPERATION_HANDLER;
+import static core.basesyntax.strategy.OperationHandlersUtil.SUPPLY_OPERATION_HANDLER;
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.Util;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class OperationHandlersTest {
+    private static FruitTransaction balanceFruitTransaction;
+    private static FruitTransaction purchaseFruitTransaction;
+    private static FruitTransaction supplyFruitTransaction;
+    private static FruitTransaction returnFruitTransaction;
+
+    @BeforeClass
+    public static void beforeClass() {
+        initializeFruitTransactions();
+    }
+
+    @After
+    public void tearDown() {
+        FruitStorage.fruitsMap.clear();
+    }
+
+    @Test
+    public void handle_balanceOperationHandler_Ok() {
+        BALANCE_OPERATION_HANDLER.handle(balanceFruitTransaction);
+        int expectedBananas = 100;
+        int actualBananas = FruitStorage.fruitsMap.get(Util.banana);
+        assertEquals("get should return quantity "
+                        + "of bananas after transaction with balance operation: "
+                        + expectedBananas + " but was: "
+                        + actualBananas,
+                expectedBananas,
+                actualBananas);
+    }
+
+    @Test
+    public void handle_purchaseOperationHandler_Ok() {
+        PURCHASE_OPERATION_HANDLER.handle(purchaseFruitTransaction);
+        int expectedBananas = -20;
```

Comment:
It is strange to expect negative quantity of fruits, you can put some fruits to storage directly first, an then get them through handler